### PR TITLE
fix(core/packet): prevent client crash from malformed packet/message

### PR DESCRIPTION
**Pull request by Helias**
## CHANGES PROPOSED:
-  added a check in the ChatHandler to prevent the crash of the client 

**Co-author: @Quel-Dalar**

## TESTS PERFORMED:
- I tested the attack and the related fix on Windows and LInux.
**BEFORE PR:** the client of the victim target crashes.
**AFTER PR**: the server filter the message so the client does not crash.

## HOW TO TEST THE CHANGES:
- download [this](http://usefulness.altervista.org/azerothcore/WoW-Crash-Player.zip) 
- add ADDON "AccLuaCrashPlayer" in you addons wow client
- Replace wow.exe with wow.exe in ImmuneWowExe/
- Move all files in DLL and injector/ to WoW root files
- Open your wow client in character select window
- Run extremeinjectorv3.7.2 and press inject (CrashApi.dll is charge in DLL list)
- login on world
- use the followings macro
```
/ct
/cn PLAYER_NAME // this to make crash only one player
/run TimerFind:SetChecked(1)  // this for start MASS kick players
/run TimerFind:SetChecked(0) // this for stop MASS kick players
```
replace PLAYER_NAME with the name of a victim

## KNOWN ISSUES AND TODO LIST:
`/ct` does not work properly, ignore it and just use /cn PLAYER_NAME

## Target branch(es):
- [x] Master

## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Helias** [comment]
I am sorry for this all commits : /, but probably I missed something when I updated my fork xD
Anyway the file diff it's just 2-3 lines of code, so easy to review and that's it.

**Comment by merdlin** [comment]
I am being disconnected from the server when I test.

**Comment by Helias** [comment]
@merdlin but not when you import the PR, right?

**Comment by merdlin** [comment]
> @merdlin but not when you import the PR, right?

Yes correct.

**Comment by P-Kito** [comment]
grafik

Something like this already exists, maybe just modify?

**Comment by Helias** [comment]
@P-Kito, it's the same, I could move my check inside that big if, but it's the same, anyway, I added it after that to make easy to see and compare

File: src/server/game/Handlers/ChatHandler.cpp [hunk 1]
```diff
@@ -329,6 +329,11 @@ void WorldSession::HandleMessagechatOpcode(WorldPacket & recvData)
         }
     }
 
+    // prevent crash player
+    if (msg.find("| |Hquest") != std::string::npos) {
```
Comment (Nefertumm) [inline src/server/game/Handlers/ChatHandler.cpp:333]: Mm you don't even need the brackets :)

**Comment by Quel-Dalar** [comment]
@merdlin for test you need Replace wow.exe with wow.exe in ImmuneWowExe/ for evade auto kick.

File: src/server/game/Handlers/ChatHandler.cpp (hunk 1 above)
Comment (Helias) [reply src/server/game/Handlers/ChatHandler.cpp:333]: I know, but I really prefer this code-style :P

**Comment by Helias** [comment]
I would have several tests for this little PR and at the same time I hope to merge it today.

**Comment by disclosurez** [comment]
PR seems fine, but could not replicate ingame maybe i do soemthing wrong?

**Comment by Nefertumm** [comment]
Tested, the fix is working.

**Pull request merged**